lib/Parser/ptree.h

```diff
@@ -338,6 +338,7 @@ struct PnFnc
     bool IsDeclaration() const { return HasFlags(kFunctionDeclaration); }
     bool IsGeneratedDefault() const { return HasFlags(kFunctionIsGeneratedDefault); }
     bool IsGenerator() const { return HasFlags(kFunctionIsGenerator); }
+    bool IsCoroutine() const { return HasFlags(kFunctionIsGenerator) || HasFlags(kFunctionIsAsync); }
```

Comment:
HasFlags accepts multiple flags, so you could do `HasFlags(kFunctionIsGenerator | kFunctionIsAsync)` instead.  Doesn't matter though.

Or at least HasFlags _should_ support multiple flags, I'm not sure if that's ever been used and thus tested